content/src/main/content/jcr_root/apps/apm/components/historyRow/historyRow.html

```diff
@@ -17,34 +17,49 @@
-    data-timeline="true" is="coral-table-row" data-script-name="${model.fileName}">
-    <td is="coral-table-cell" coral-table-rowselect>
-    </td>
-    <td is="coral-table-cell" value="${model.fileName}">
-        ${model.fileName}
-    </td>
-    <td is="coral-table-cell" value="${model.author}">
-        ${model.author}
-    </td>
-    <td is="coral-table-cell" value="${model.uploadTime.timeInMillis || 0}">
-        <time datetime="${model.uploadTime.timeInMillis || 0}"
-              data-sly-use.modified="${'com.adobe.cq.xf.ui.DateFormatter' @ date=model.uploadTime, relativeFormat='r'}">
-            ${modified.uploadTime}
-        </time>
-    </td>
-    <td is="coral-table-cell" value="${model.executionTime.timeInMillis || 0}">
-        <time datetime="${model.executionTime.timeInMillis || 0}"
-              data-sly-use.modified="${'com.adobe.cq.xf.ui.DateFormatter' @ date=model.v, relativeFormat='r'}">
-            ${modified.executionTime}
-        </time>
-    </td>
-    <!--<td is="coral-table-cell" value="${model.lastDryRun.timeInMillis || 0}">-->
-        <!--<time datetime="${model.lastDryRun.timeInMillis || 0}"-->
-              <!--data-sly-use.modified="${'com.adobe.cq.xf.ui.DateFormatter' @ date=model.lastDryRun, relativeFormat='r'}">-->
-            <!--${modified.lastDryRun}-->
-        <!--</time>-->
-    <!--</td>-->
-</tr>
+<sly data-sly-use.item="com.cognifide.cq.cqsm.api.history.Entry"/>
+<tr class="foundation-collection-item foundation-collection-navigator"
+    data-foundation-collection-item-id="${resource.path}" data-path="${resource.path}"
+    data-timeline="true" is="coral-table-row" data-script-name="${item.fileName}">
+
+  <td is="coral-table-cell" coral-table-rowselect>
+    <coral-icon icon="fileKey" size="M">
+    </coral-icon>
+  </td>
+  <td is="coral-table-cell" value="${item.fileName}">
+    <a href="${item.filePath}" download="${item.fileName}">${item.fileName}</a>
+  </td>
+  <td is="coral-table-cell" value="${item.author}">
+    ${item.author}
+  </td>
+  <td is="coral-table-cell" value="${item.lastModified.time || 0}">
+    <time class="text-uppercase text-muted" data-sly-test="${item.lastModified}"
+          datetime="${item.lastModified.time || 0}">
+      ${'MMM dd, yyyy hh:mm:ss a' @ format=item.lastModified, locale=request.locale.language}
+    </time>
+  </td>
+  <td is="coral-table-cell" value="${item.executor}">
+    ${item.executor}
+  </td>
+  <td is="coral-table-cell" value="${item.executionTime.time || 0}">
+    <sly data-sly-test="${item.executionTime && item.instanceType}">
+      <img class="${item.isRunSuccessful ? 'run-success' : 'run-error'}"/>
+      <span class="last-execution-status">
+        <span class="instance">${item.instanceType}</span> <time datetime="${item.executionTime.time || 0}">${'MMM dd, yyyy hh:mm:ss a' @ format=item.executionTime, locale=request.locale.language, type='date'}</time>

```

Comment:
can we move the formatting logic to the model or some other class? It's being duplicated at the line 61. Besides, having such logic in the markup is not the best approach which we could take

Reply:
Ok, I moved formatting logic to Java class, thanks.